components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/UserStoreConfigAdminService.java

```diff
@@ -527,136 +320,54 @@ private boolean validateDomainsForDelete(String[] domains) {
-                        if (randomPassword != null) {
-                            if (propertyDTOValue.equalsIgnoreCase(randomPassword.getRandomPhrase())) {
-                                propertyDTOValue = randomPassword.getPassword();
-                            }
-                        }
-                    }
+       validateDomain(domain, isDisable);
+       UserStoreDTO userStoreDTO = getUserStoreDTO(domain, isDisable, null);
+       updateTheStateInFileRepository(userStoreDTO);
+   }
 
-                    try {
-                        propertyDTOValue = SecondaryUserStoreConfigurationUtil.encryptPlainText(propertyDTOValue);
-                        encrypted = true;
-                    } catch (IdentityUserStoreMgtException e) {
-                        log.error("addProperties failed to encrypt", e);
-                        //its ok to continue from here
-                    }
-                }
-                addProperty(propertyDTOName, propertyDTOValue, doc, parent, encrypted);
-            }
+    private void updateTheStateInFileRepository(UserStoreDTO userStoreDTO) throws IdentityUserStoreMgtException {
+
+        try {
+            getFileBasedUserStoreDAOFactory().updateUserStore(userStoreDTO, true);
+        } catch (Exception e) {
+            String errorMessage = e.getMessage();
+            throw new IdentityUserStoreMgtException(errorMessage);
         }
     }
 
     /**
-     * Adds a property
-     *
-     * @param name:   Name of property
-     * @param value:  Value
-     * @param doc:    Document
-     * @param parent: Parent element of the property to be added as a child
+     * Update the status of domain.
+     * @param domain userstore domain
+     * @param isDisable true if the userstore domain is disabled.
+     * @param repositoryClass repository class
+     * @throws IdentityUserStoreMgtException throws an error when changing the status of the user store.
      */
-    private void addProperty(String name, String value, Document doc, Element parent, boolean encrypted) {
-        Element property = doc.createElement("Property");
-        Attr attr;
-        if (encrypted) {
-            attr = doc.createAttribute("encrypted");
-            attr.setValue("true");
-            property.setAttributeNode(attr);
-        }
-
-        attr = doc.createAttribute("name");
-        attr.setValue(name);
-        property.setAttributeNode(attr);
-
-        property.setTextContent(value);
-        parent.appendChild(property);
-    }
-
-
-    private void deleteFile(File file, final String userStoreName) throws IdentityUserStoreMgtException {
-        if (!IdentityUtil.isValidFileName(userStoreName)) {
-            String message = "Provided domain name : '" + userStoreName + "' is invalid.";
-            log.error(message);
-            throw new IdentityUserStoreMgtException(message);
-        }
+    public void enableDisableUserStore(String domain, Boolean isDisable, String repositoryClass)
```

Comment:
If there is a limitation to use changeUserStoreState, shall we use modifyUserStoreState